src/shared/awsContext.ts

```diff
@@ -55,6 +59,40 @@ export class DefaultAwsContext implements AwsContext {
         const persistedRegions = context.globalState.get<string[]>(regionSettingKey)
         this.explorerRegions = persistedRegions || []
     }
+    /**
+    
+        Checks in order of precedence
+       1. explorerRegion[0] if only one region exists.
+       2. the node passed to the current command
+       3. last-expanded AWS Explorer service / etc
+       4. last selection region in RegionProvider of Wizard. 
+       5. final fallback is the credentials default region
+
+     * @param node node on current command. 
+     * @returns heuristic for default region.
+     */
+    public guessDefaultRegion(node?: AWSTreeNodeBase): string {
+        if (this.explorerRegions.length === 1) {
+            return this.explorerRegions[0]
+        } else if (node?.regionCode) {
+            return node.regionCode
```

Comment:
This should be the first choice, because it's an explicit arg. Even though in practice it likely never matters.